src/improvements/SuperchainAddressRegistry.sol

```diff
@@ -210,6 +214,13 @@ contract SuperchainAddressRegistry is StdChains {
     // ======== Superchain address discovery functions ========
     // ========================================================
 
+    /// @notice After instantiation of this contract, you can continue to discover new chains by calling this function.
+    /// This makes addresses on these chains available by using the getAddress function.
+    function discoverNewChain(ChainInfo memory chain) public {
```

Comment:
We should update the tests for this contract to also write a new chain before reading to assert we don't accidentally overwrite or modify anything when calling this

Reply:
done: https://github.com/ethereum-optimism/superchain-ops/pull/972/commits/7cf7034df87b699639fa18fb0fe8509ca64ee264